pkg/ceph/rgw/objectstore.go

```diff
@@ -64,12 +64,23 @@ func CreateObjectStore(context *Context, metadataSpec, dataSpec model.Pool, serv
 }
 
 func DeleteObjectStore(context *Context) error {
-	err := deleteRealm(context)
+	stores, err := GetObjectStores(context)
+	if err != nil {
+		return fmt.Errorf("failed to detect object stores during deletion. %+v", err)
+	}
+	logger.Infof("Found stores %v when deleting store %s", stores, context.Name)
+
+	err = deleteRealm(context)
 	if err != nil {
 		return fmt.Errorf("failed to delete realm. %+v", err)
 	}
 
-	err = deletePools(context)
+	lastStore := false
+	if len(stores) == 1 && stores[0] == context.Name {
```

Comment:
what would it mean if `len(stores) == 1 && stores[0] != context.Name`?  someone is trying to delete a store that doesn't exist?

Reply:
you are correct. I don't know how this could happen, but wanted this check to make sure it's a valid request